src/components/AttachmentModal.js

```diff
@@ -110,14 +111,19 @@ class AttachmentModal extends PureComponent {
 
     /**
      * Execute the onConfirm callback and close the modal.
+     *
+     * @param {Event} event Either a click event, keyboard event or undefined
      */
-    submitAndClose() {
+    submitAndClose(event) {
         // If the modal has already been closed, don't allow another submission
         if (!this.state.isModalOpen) {
             return;
         }
-
         if (this.props.onConfirm) {
+            // Ignore the submit/close if it was initiated by enter key and send-on-enter is disabled
+            if (!this.state.isSendOnEnterEnabled && (!event || event instanceof KeyboardEvent)) {
```

Comment:
Yes, this was already discussed. We need this to prevent send **if the user submits the password with the enter key**.

Please see the comments and video above: **premature_send_with_enter_key.mp4**.

Reply:
I don't see where it was confirmed. Enter should trigger CTA which is `send Attachment`. @shawnborton What do you think the behavior should be? 
- Pressing Enter while user is typing password.
- Pressing Enter normally. 

I think that in both cases it should trigger `Send attachment`. It does not matter if user has entered the password or not for sending that attachment.

Reply:
I believe this was confirmed by @Santhosh-Sellavel:

> Yeah that case handle it please, press enter to send doesn't work breaks other attachments too.

In any case...
1. User enters an invalid password
2. User presses the enter key **expecting to see the PDF**
3. However the user doesn't see the PDF
4. Instead, the form submits and the PDF is sent - which was **not what the user expected** when he pressed the enter key

In my opinion, that's a terrible user experience.

Reply:
I would think that if you are focused into the password field and you hit enter, then we should submit the password form and not send the attachment.

Reply:
@frenkield Let's do this as Shawn said.

Next, Let's say user didn't enter any password and the upload preview is open. Pressing Enter does nothing at the moment. Shouldn't it send the attachment?